sui/deploy-contract.js

```diff
@@ -5,156 +5,206 @@ const { toB64 } = require('@mysten/sui/utils');
+        });
+    }
 
-            const tx = new Transaction();
+    const result = await broadcast(client, keypair, tx);
 
-            const separator = tx.moveCall({
-                target: `${packageId}::bytes32::new`,
-                arguments: [tx.pure(arrayify(domainSeparator))],
-            });
+    printInfo('Setup Gateway', result.digest);
 
-            tx.moveCall({
-                target: `${packageId}::gateway::setup`,
-                arguments: [
-                    tx.object(creatorCap),
-                    tx.pure.address(operator),
-                    separator,
-                    tx.pure.u64(minimumRotationDelay),
-                    tx.pure.u64(options.previousSigners),
-                    tx.pure(bcs.vector(bcs.u8()).serialize(encodedSigners).toBytes()),
-                    tx.object(suiClockAddress),
-                ],
-            });
+    const [gateway] = getObjectIdsByObjectTypes(result, [`${packageId}::gateway::Gateway`]);
 
-            if (policy !== 'any_upgrade') {
-                const upgradeType = policy === 'code_upgrade' ? 'only_additive_upgrades' : 'only_dep_upgrades';
+    // Update chain configuration
+    chain.contracts.AxelarGateway = {
+        objects: {
+            Gateway: gateway,
+            RelayerDiscovery: relayerDiscovery,
+            UpgradeCap: upgradeCap,
+        },
+        domainSeparator,
+        operator,
+        minimumRotationDelay,
+    };
+}
 
-                tx.moveCall({
-                    target: `${suiPackageAddress}::package::${upgradeType}`,
-                    arguments: [tx.object(upgradeCap)],
-                });
-            }
+async function deploy(keypair, client, supportedContract, config, chain, options) {
+    const { packageDir, packageName } = supportedContract;
 
-            const result = await broadcast(client, keypair, tx);
+    const published = await deployPackage(packageDir, client, keypair, options);
 
-            printInfo('Setup transaction digest', result.digest);
+    printInfo(`Deployed ${packageName}`, published.publishTxn.digest);
 
-            const [gateway] = getObjectIdsByObjectTypes(result, [`${packageId}::gateway::Gateway`]);
+    chain.contracts[packageName] = {
+        address: published.packageId,
+    };
 
-            contractConfig.objects = {
-                gateway,
-                relayerDiscovery,
-                upgradeCap,
-            };
-            contractConfig.domainSeparator = domainSeparator;
-            contractConfig.operator = operator;
-            contractConfig.minimumRotationDelay = minimumRotationDelay;
+    switch (packageName) {
+        case 'GasService':
+            await postDeployGasService(published, chain);
             break;
-        }
-
-        default: {
-            throw new Error(`${contractName} is not supported.`);
-        }
+        case 'AxelarGateway':
+            await postDeployAxelarGateway(published, keypair, client, config, chain, options);
```

Comment:
define a map from contract name to postDeploy script. Avoids the switch case, and standardize input args